### PR TITLE
Update vtgatev2 test to better support testing gateway implementations.

### PR DESCRIPTION
**Pull request by guoliang100**
@alainjobart @sougou

### CODE REVIEW

File: test/topo_flavor/etcd.py [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
 
 
-class EtcdCluster:
+class EtcdCluster(object):
   """Sets up a global or cell-local etcd cluster."""
 
   def __init__(self, name):
-    import environment
-    import utils
+    import environment  # pylint: disable=g-import-not-at-top
```
Comment (alainjobart) [inline test/topo_flavor/etcd.py]: This doesn't look right, please remove all pylint comments in this file related to imports, they should be on the top...

File: test/topo_flavor/zookeeper.py [hunk 2]
```diff
@@ -1,30 +1,32 @@
...
-    """Assign ports if not already assigned"""
+    """Assign ports if not already assigned."""
 
     if self.ports_assigned:
       return
 
-    from environment import reserve_ports
-    import utils
+    from environment import reserve_ports  # pylint: disable=g-import-not-at-top
+    import utils  # pylint: disable=g-import-not-at-top
```
Comment (alainjobart) [inline test/topo_flavor/zookeeper.py]: same here, remove the pylint comments, the warning is correct.

File: test/vtgatev2_test.py [hunk 3]
```diff
@@ -1050,34 +1148,47 @@ def test_restart_mysql_tablet_idle(self):
...
-      self.fail(
-          'DatabaseError should have been raised, but got %s' % str(result))
-    except Exception, e:  # pylint: disable=broad-except
-      self.assertIsInstance(e, dbexceptions.DatabaseError)
-      self.assertNotIsInstance(e, dbexceptions.IntegrityError)
-      self.assertNotIsInstance(e, dbexceptions.OperationalError)
-      self.assertNotIsInstance(e, dbexceptions.TimeoutError)
+
+    if (protocols_flavor().tabletconn_protocol() != 'grpc'
+        and vtgate_gateway_flavor().flavor() == 'shardgateway'):
```
Comment (alainjobart) [inline test/vtgatev2_test.py]: Can you please document this restriction? The gRPC one is mentioned below, the shardgateway one might be surprising.

**Comment by alainjobart** [comment]
LGTM after comments are addressed.

File: test/topo_flavor/etcd.py (hunk 1 above)
Comment (guoliang100) [inline test/topo_flavor/etcd.py]: These are circular dependences. They cannot be placed on the top.

File: test/topo_flavor/zookeeper.py (hunk 2 above)
Comment (guoliang100) [inline test/topo_flavor/zookeeper.py]: Same here.

File: test/vtgatev2_test.py (hunk 3 above)
Comment (guoliang100) [inline test/vtgatev2_test.py]: Updated the comment.

**Comment by guoliang100** [comment]
I will merge the PR. We can follow up on the imports separately.

**Pull request merged**